File: 1287. Element Appearing More Than 25% In Sorted Array.cpp
class Solution {
public:
    int findSpecialInteger(vector<int>& arr) {
        int count = 1;
        for (int i = 1; i < arr.size(); i++) {
            if (arr[i] == arr[i - 1]) {
                count++;
            } else {
                count = 1;
            };
            if (count > arr.size() / 4) {
                return arr[i];
            };
        };
        return arr.back();
    };
};

File: 1287. Element Appearing More Than 25% In Sorted Array.py
class Solution:
    def findSpecialInteger(self, arr: List[int]) -> int:
        count = 0
        for i, num in enumerate(arr):
            if num != arr[i - 1]:
                count = 1
            else:
                count += 1
            if count > len(arr) / 4:
                return num
        return arr[-1]
